jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -397,41 +395,81 @@ private static void ensureProperties() {
 		if (key == null) {
 			break;
 		}
-		systemProperties.put(key, list[i+1]);
+		initializedProperties.put(key, list[i+1]);
 	}
-	
-	propertiesInitialized = true;
-	
-	/*[IF Java12]*/
-	/* java.lang.VersionProps.init() eventually calls into System.setProperty() where propertiesInitialized needs to be true */
-	java.lang.VersionProps.init(systemProperties);
-	/*[ELSE]
-	/*[IF Sidecar19-SE]*/
+
 	/* java.lang.VersionProps.init() eventually calls into System.setProperty() where propertiesInitialized needs to be true */
+	propertiesInitialized = true;
+
+/*[IF Java12]*/
+	java.lang.VersionProps.init(initializedProperties);
+/*[ELSE]
+	/* VersionProps.init requires systemProperties to be set */
+	systemProperties = initializedProperties;
+
+/*[IF Sidecar19-SE]*/
 	java.lang.VersionProps.init();
-	/*[ELSE]*/
+/*[ELSE]
 	sun.misc.Version.init();
-	/*[ENDIF] Sidecar19-SE */
-	/*[ENDIF] Java12 */
 
-	/*[IF !Sidecar19-SE]*/
 	StringBuffer.initFromSystemProperties(systemProperties);
 	StringBuilder.initFromSystemProperties(systemProperties);
-	/*[ENDIF]*/
+/*[ENDIF] Sidecar19-SE */
+/*[ENDIF] Java12 */
 
-	String javaRuntimeVersion = systemProperties.getProperty("java.runtime.version"); //$NON-NLS-1$
+/*[IF Java12]*/
+	String javaRuntimeVersion = initializedProperties.get("java.runtime.version"); //$NON-NLS-1$
+/*[ELSE]
+	String javaRuntimeVersion = initializedProperties.getProperty("java.runtime.version"); //$NON-NLS-1$
+/*[ENDIF] Java12 */
 	if (null != javaRuntimeVersion) {
-		String fullVersion = systemProperties.getProperty("java.fullversion"); //$NON-NLS-1$
+	/*[IF Java12]*/
+		String fullVersion = initializedProperties.get("java.fullversion"); //$NON-NLS-1$
+	/*[ELSE]
+		String fullVersion = initializedProperties.getProperty("java.fullversion"); //$NON-NLS-1$
+	/*[ENDIF] Java12 */
 		if (null != fullVersion) {
-			systemProperties.put("java.fullversion", (javaRuntimeVersion + "\n" + fullVersion)); //$NON-NLS-1$ //$NON-NLS-2$
+			initializedProperties.put("java.fullversion", (javaRuntimeVersion + "\n" + fullVersion)); //$NON-NLS-1$ //$NON-NLS-2$
 		}
 		rasInitializeVersion(javaRuntimeVersion);
 	}
 
-	lineSeparator = systemProperties.getProperty("line.separator", "\n"); //$NON-NLS-1$
+/*[IF Java12]*/
+	lineSeparator = initializedProperties.getOrDefault("line.separator", "\n"); //$NON-NLS-1$
+/*[ELSE]
+	lineSeparator = initializedProperties.getProperty("line.separator", "\n"); //$NON-NLS-1$
+/*[ENDIF] Java12 */
 	/*[IF Sidecar19-SE]*/
-	useLegacyVerPresents = systemProperties.containsKey("use.legacy.version");
+	useLegacyVerPresents = initializedProperties.containsKey("use.legacy.version");
 	/*[ENDIF]*/
+
+	if (isInitialization) {
+		/*[PR CMVC 179976] System.setProperties(null) throws IllegalStateException */
+		/*[IF Java12]*/
+			VM.saveProperties(initializedProperties);
```

Comment:
nit: please remove the indentation (that _was_ appropriate before the try-catch was removed).

Reply:
woops. updated